src/OpenTelemetry.Extensions.Hosting/Implementation/SelfDiagnosticsLoggingHostedService.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="SelfDiagnosticsLoggingHostedService.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Extensions.Hosting.Implementation
+{
+    internal class SelfDiagnosticsLoggingHostedService : IHostedService
+    {
+        private readonly IServiceProvider serviceProvider;
+        private IDisposable forwarder;
+
+        public SelfDiagnosticsLoggingHostedService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // The sole purpose of this HostedService is to
+            // start forwarding the self-diagnostics events
+            // to the logger factory
+            this.forwarder = new SelfDiagnosticsEventLogForwarder(System.Diagnostics.Tracing.EventLevel.LogAlways, this.serviceProvider.GetService<ILoggerFactory>());
```

Comment:
`LogAlways` here means all the logs are going to collected by the forwarder.... Won't it mean checks like this (https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs#L95), would always be true and causes perf issues?

Reply:
The messages get filtered out in the logger and can be controlled by logging configuration. The same logging system used in all of .NET Core production apps. I could make this a parameter going into the addselfdiagnostics method, but that would mean you would need to manually set that value and changing it in your logging configuration might not have any affect.

Reply:
The specification calls out having a dedicated option for controlling log level, so I think being able to adjust that independently of an application's logging level makes sense. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration

Reply:
Even if the ILogger logs are configured for ,say, only Error level, the current code subscribes to *all level* EventsSource logs (and drop the ones below Error.). This'd affect perf.

Reply:
@cijothomas I just pushed changes that set the EventLevels based on the logging configuration. It will also update them if the logging configuration changes at run time.

Reply:
Thanks. Will re-review this part.